src/rta/fullsupernodelist.cpp

```diff
@@ -0,0 +1,352 @@
+{
+
+    if (exists(item->walletAddress()))
+        return false;
+
+    boost::unique_lock<boost::shared_mutex> writerLock(m_access);
+    m_list.insert(std::make_pair(item->walletAddress(), SupernodePtr{item}));
+    LOG_PRINT_L0("added supernode: " << item->walletAddress());
+    LOG_PRINT_L0("list size: " << m_list.size());
+    return true;
+}
+
+size_t FullSupernodeList::loadFromDir(const string &base_dir)
+{
+    vector<string> wallets = findWallets(base_dir);
+    size_t result = 0;
+    LOG_PRINT_L0("found wallets: " << wallets.size());
+    for (const auto &wallet_path : wallets) {
+        Supernode * sn = Supernode::load(wallet_path, "", m_daemon_address, m_testnet);
+        if (sn)  {
+            if (!this->add(sn)) {
+                LOG_ERROR("Can't add supernode " << sn->walletAddress() << ", already exists");
+                delete sn;
+            } else {
+                LOG_PRINT_L1("Added supernode: " << sn->walletAddress() << ", stake: " << sn->stakeAmount());
+                ++result;
+            }
+        }
+    }
+    return result;
+}
+
+
+
+size_t FullSupernodeList::loadFromDirThreaded(const string &base_dir, size_t &found_wallets)
+{
+    vector<string> wallets = findWallets(base_dir);
+    LOG_PRINT_L0("found wallets: " << wallets.size());
+    found_wallets = wallets.size();
+
+    utils::ThreadPool tp;
+
+    auto worker = [&](const std::string &wallet_path) {
+        Supernode * sn = Supernode::load(wallet_path, "", m_daemon_address, m_testnet);
+        if (sn) {
+            if (!this->add(sn)) {
+                LOG_ERROR("can't add supernode: " << sn->walletAddress());
+                delete sn;
+            } else {
+                LOG_PRINT_L0("Loaded supernode: " << sn->walletAddress());
+            }
+        } else {
+            LOG_ERROR("Error loading supernode wallet from: " << wallet_path);
+        }
+    };
+
+    for (const auto &wallet_path : wallets) {
+        tp.enqueue(boost::bind<void>(worker, wallet_path));
+    }
+    tp.run();
+    return this->size();
+}
+
+bool FullSupernodeList::remove(const string &address)
+{
+    boost::unique_lock<boost::shared_mutex> readerLock(m_access);
+    m_list.erase(address) > 0;
+}
+
+size_t FullSupernodeList::size() const
+{
+    boost::shared_lock<boost::shared_mutex> readerLock(m_access);
+    return m_list.size();
+}
+
+bool FullSupernodeList::exists(const string &address) const
+{
+
+    boost::shared_lock<boost::shared_mutex> readerLock(m_access);
+    return m_list.find(address) != m_list.end();
+}
+
+bool FullSupernodeList::update(const string &address, const std::vector<Supernode::KeyImage> &key_images)
+{
+
+    boost::unique_lock<boost::shared_mutex> writerLock(m_access);
+    auto it = m_list.find(address);
+    if (it != m_list.end()) {
+        return it->second->importKeyImages(key_images);
+    }
+    return false;
+}
+
+FullSupernodeList::SupernodePtr FullSupernodeList::get(const string &address) const
```

Comment:
Check all method, I guess the problem with method autogeneration: almost all methods use `string`, not `std::string`

Reply:
fixed in 9244de51da07fa9cb844efc850ad6649a7bc5dd8, removed "std" prefix in cpp file for all stl classes